`import spikeinterface.extractors` fails if NWB not installed

Hey,

The typing here creates a crash if NWB is not installed https://github.com/SpikeInterface/spikeinterface/blob/72b0835cd855cb3fac165a362646a7312c0c73f3/src/spikeinterface/extractors/nwbextractors.py#L40

```
$ pytest
ImportError while loading conftest '/home/aurelien/Documents/lussac2/tests/conftest.py'.
tests/conftest.py:6: in <module>
    from lussac.core import LussacData, LussacPipeline, MonoSortingData, MultiSortingsData
src/lussac/core/__init__.py:1: in <module>
    from .lussac_data import LussacData, MonoSortingData, MultiSortingsData
src/lussac/core/lussac_data.py:12: in <module>
    import spikeinterface.extractors as se
../../dev/spikeinterface/spikeinterface/src/spikeinterface/extractors/__init__.py:1: in <module>
    from .extractorlist import *
../../dev/spikeinterface/spikeinterface/src/spikeinterface/extractors/extractorlist.py:16: in <module>
    from .nwbextractors import (NwbRecordingExtractor, NwbSortingExtractor,
../../dev/spikeinterface/spikeinterface/src/spikeinterface/extractors/nwbextractors.py:40: in <module>
    def retrieve_electrical_series(nwbfile: NWBFile, electrical_series_name: Optional[str] = None) -> ElectricalSeries:
E   NameError: name 'NWBFile' is not defined
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -77,7 +77,7 @@ streaming_extractors = [
     "fsspec", 
     "aiohttp",
     "requests",
-    "pynwb",
+    "pynwb>=2.3.0",
 ]
 
 full = [

--- a/src/spikeinterface/extractors/nwbextractors.py
+++ b/src/spikeinterface/extractors/nwbextractors.py
@@ -1,23 +1,25 @@
+from __future__ import annotations
 from pathlib import Path
-from typing import Union, List, Optional, Literal
+from typing import Union, List, Optional, Literal, Dict
 
 import numpy as np
-import h5py
 
 from spikeinterface import get_global_tmp_folder
 from spikeinterface.core import BaseRecording, BaseRecordingSegment, BaseSorting, BaseSortingSegment
 from spikeinterface.core.core_tools import define_function_from_class
 
 try:
-    import pynwb
     from pynwb import NWBHDF5IO, NWBFile
-    from pynwb.ecephys import ElectricalSeries, FilteredEphys, LFP, ElectrodeGroup
-    from hdmf.data_utils import DataChunkIterator
-    from hdmf.backends.hdf5.h5_utils import H5DataIO
-
+    from pynwb.ecephys import ElectricalSeries, ElectrodeGroup
     HAVE_NWB = True
 except ModuleNotFoundError:
     HAVE_NWB = False
+
+try:
+    import h5py
+    HAVE_H5PY = True
+except:
+    HAVE_H5PY = False
 
 try:
     import fsspec

